lib/bowl.rb

```diff
@@ -0,0 +1,71 @@
+class BowlingScorer
+  def initialize
+    @player_score = 0
+    @frames = []
+  end
+
+  def add_frame(shot1, shot2)
+    fail "Looks like you hit the next lane or something, cant knock more than 10" if shot1 + shot2 > 10
+    fail "Smells like invalid input" if shot1 < 0 || shot2 < 0
+    ## Line below is irrelevant for the current specs, but this is done assuming when UI is developed
+    ## the user will input their shots sequentially as opposed to passing them in together
+    shot1 == 10 ? current_frame = [10,0] : current_frame = [shot1, shot2]
+    @frames << current_frame
+    return current_frame
+  end
+
+  def check_for_specials(frame)
+    message = ""
+    if frame[0] == 10
+      message = "strike"
+    elsif frame.sum == 10
+      message = "spare"
+    end
+    return message
+  end
+
+  def add_bonus_frame(*args)
```

Comment:
Based on the focus points provided to us I would suggest encapsulating the Frame logic into a seperate class.  This would make it more modular and the bowlingscorer class would be only responsible for managing the Frames rather than the logic.